clang/test/SemaHLSL/resource_binding_attr_error_resource.hlsl

```diff
@@ -0,0 +1,71 @@
+// RUN: %clang_cc1 -triple dxil-pc-shadermodel6.3-library -x hlsl -o - -fsyntax-only %s -verify
+
+// This test validates the diagnostics that are emitted when a variable with a "resource" type
+// is bound to a register using the register annotation
+
+/*
+template<typename T>
+struct [[hlsl::resource_class(SRV)]] MyTemplatedSRV {
+  T x;
+};
+
+struct [[hlsl::resource_class(SRV)]] MySRV {
+  int x;
+};
+
+struct [[hlsl::resource_class(Sampler)]] MySampler {
+  int x;
+};
+
+struct [[hlsl::resource_class(UAV)]] MyUAV {
+  int x;
+};
+
+struct [[hlsl::resource_class(CBuffer)]] MyCBuffer {
+  int x;
+};
+*/ 

```

Comment:
Forgot to remove the commented block.
Correct, there are currently no examples that match that look. I believe this is something I expect @hekota to change (applying attributes to the handle type rather than the handle variable), and then I can retroactively update these tests.

Reply:
It's a shame to be checking in code that we know will need to be changed shortly after.  At the very least, please file an issue to track updating these tests.

Reply:
https://github.com/llvm/wg-hlsl/issues/43